subworkflows/local/binning.nf

```diff
@@ -0,0 +1,153 @@
+    }
+
+    SPLIT_FASTA ( ch_input_splitfasta )
+
+    // decompress main bins (and large unbinned contigs from SPLIT_FASTA) for
+    // MAG_DEPTHS, first have to separate and re-group due to limitation of
+    // GUNZIP module
+    METABAT2_METABAT2.out.fasta.transpose().set { ch_metabat2_results_transposed }
+    MAXBIN2.out.binned_fastas.transpose().set { ch_maxbin2_results_transposed }
+    SPLIT_FASTA.out.unbinned.transpose().set { ch_split_fasta_results_transposed }
+
+    ch_metabat2_results_transposed
+        .mix( ch_maxbin2_results_transposed, ch_split_fasta_results_transposed )
+        .set { ch_final_bins_for_gunzip }
+
+    GUNZIP ( ch_final_bins_for_gunzip )
+    GUNZIP.out.gunzip
+        .set{ ch_binning_results_gunzipped }
+
+    // Compute bin depths for different samples (according to `binning_map_mode`)
+    // Have to remove binner meta for grouping to mix back with original depth
+    // files, as required for MAG_DEPTHS
+    // Q: where is maxbin2 noclass? All filtered out already?
+    ch_binning_results_gunzipped
+            .map { meta, results ->
+                def meta_new = meta.clone()
+                [ [ 'id': meta_new['id'], 'group': meta_new['group'], 'single_end': meta_new['single_end'], 'assembler': meta_new['assembler'] ], results ]
+            }
+        .groupTuple (by: 0 )
+        .join( METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS.out.depth, by: 0 )
+        .set { ch_depth_input }
+
+    MAG_DEPTHS ( ch_depth_input )
+
+    // Plot bin depths heatmap for each assembly and mapped samples (according to `binning_map_mode`)
+    // create file containing group information for all samples
+    ch_sample_groups = reads
+        .collectFile(name:'sample_groups.tsv'){ meta, reads -> meta.id + '\t' + meta.group + '\n' }
+
+    // filter MAG depth files: use only those for plotting that contain depths for > 2 samples
+    ch_mag_depths_plot = MAG_DEPTHS.out.depths
+        .map { meta, depth_file -> if (getColNo(depth_file) > 2) [meta, depth_file] }
+
+    MAG_DEPTHS_PLOT ( ch_mag_depths_plot, ch_sample_groups.collect() )
+    MAG_DEPTHS_SUMMARY ( MAG_DEPTHS.out.depths.map{it[1]}.collect() )
+
+    // Group final binned contigs per sample for final output
+    ch_binning_results_gunzipped
+        .groupTuple(by: 0)
+        .set{ ch_binning_results_final }
+
+    emit:
+    bins                                         = ch_binning_results_final
+    unbinned                                     = SPLIT_FASTA.out.unbinned
+    tooshort                                     = METABAT2_METABAT2.out.tooshort
+    lowdepth                                     = METABAT2_METABAT2.out.lowdepth
+    depths_summary                               = MAG_DEPTHS_SUMMARY.out.summary
+    metabat2_version                             = METABAT2_METABAT2.out.versions
+    metabat2_jgisummarizebamcontigdepths_version = METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS.out.versions
```

Comment:
Is the Maxbin2 version reported somewhere? I would have thought here it should appear (and in `mag.nf` where I also cannot finnd it)

Reply:
Oops, good Catch"